core/src/main/scala/io/finch/request/package.scala

```diff
@@ -309,7 +358,9 @@ package object request extends LowPriorityImplicits {
    *
    * @param message the message
    */
-  class RequestReaderError(val message: String) extends Exception(message)
+  class RequestReaderError(val message: String, cause: Throwable) extends Exception(message, cause) {
+    def this(message: String) = this(message, null)
```

Comment:
I think there is actually no way to get rid of the null. Even if we'd use Option, at the point we pass it to the Java constructor, we'd need to call `cause.getOrElse(null)`, as Java does not know about Options. And you want to pass the cause up somehow, to stick to the contract of the Java Exception, and to have the cause included when you call `getStacktrace`.